engine/battle/init_battle_variables.asm

```diff
@@ -20,12 +20,16 @@ InitBattleVariables:
 	ld [hli], a ; wPlayerHPBarColor
 	ld [hl], a ; wEnemyHPBarColor
 	ld hl, wCanEvolveFlags
-	ld b, $3c
-.loop
+	ld b, $3c ; What is this?
```

Comment:
This is simply setting $3c bytes of wram starting from wCanEvolveFlags to a value of zero. You can probably do the following:

```suggestion
	ld b, wMiscBattleDataEnd - wMiscBattleData
```

Then add `wMiscBattleData::` above `wCanEvolveFlags` and add `wMiscBattleDataEnd` below the `ds 8` free bytes here:
```
wEnemyBideAccumulatedDamage:: dw

	ds 8
```

Reply:
Thank you for explaining this! 😃